slave/process-hadoop-hbase/bin/process-hadoop_hbase.sh

```diff
@@ -0,0 +1,123 @@
+#! /bin/bash
+
+#
+# Manage HBase user namespaces in Hadoop environment
+#
+# In addition unix users and groups need to be managed on the whole Hadoop cluser.
+#
+# Where to launch:
+# * on master node (or any node with HBase client and admin client keytab)
+# * launch only once
+#
+# Requirements:
+# * client admin Kerberos keytab (not required for unsecured cluster)
+# * HBase client installed and configured
+#
+# Physical delete:
+# * disabled by default (enable using FORCE_DELETE=1)
+# * removes all the tables in user namespace and the namespace itself
+#
+# Last actions are stored in /var/tmp/perun-hadoop-hbase-*.sh.
+#
+# Usernames can't contain '-' character, they will by skipped.
+#
+
+PROTOCOL_VERSION='3.0.0'
+
+function process() {
+	FROM_PERUN="${WORK_DIR}/hadoop_hbase"
+
+	FORCE_DELETE=${FORCE_DELETE:-'0'}
+	KEYTAB=${KEYTAB:-'/etc/security/keytab/hbase.service.keytab'}
+	PRINCIPAL=${PRINCIPAL:-"hbase/`hostname -f`@${REALM}"}
+	RESERVED=${RESERVED:-'NAMESPACE|default|hbase|hive|oozie'}
+
+	KRB5CCNAME="FILE:${WORK_DIR}/krb5cc_perun_hadoop_hbase"
+
+	I_HBASE_CREATED=(0 'User ${login} will be created.')
+	I_HBASE_DELETED=(0 'User ${login} will be deleted.')
+	I_HBASE_OK=(0 'HBase shell OK.')
+	I_HBASE_SIMULATED=(0 'Delete actions not performed.')
+	E_KINIT_FAILED=(1 'Kinit on HBase master failed')
+	E_HBASE_NAMESPACE_LIST=(2 'Cannot get list of namespaces')
+	E_HBASE_TABLE_LIST=(3 'Cannot get list of tables from ${login}')
+	E_HBASE_FAILED=(4 'HBase shell failed')
+	E_EMPTY_LIST=(5 'The list is empty!')
+	E_EMPTY_USERNAME=(6 'Empty username')
+
+	create_lock
+	chown hbase "${WORK_DIR}"
```

Comment:
There is no need for recursive chown on whole work directory? What about files?

Reply:
I guess this was needed in the past to access to the work directory for other users. The Hadoop service command are launched under 'hdfs' and 'hbase' users.

But I've just tested it without the changing owner an it works fine, so it could be removed...